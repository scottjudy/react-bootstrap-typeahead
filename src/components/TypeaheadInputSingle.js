// @flow

import React from 'react';

import Hint, { type ShouldSelect } from './Hint';
import Input from './Input';

import withClassNames from '../behaviors/classNames';

import type { RefCallback, ReferenceElement } from '../types';

type Props = {
  inputRef: RefCallback<HTMLInputElement>,
  referenceElementRef: RefCallback<ReferenceElement>,
  shouldSelectHint?: ShouldSelect,
};

export default withClassNames(
  ({ inputRef, referenceElementRef, shouldSelectHint, ...props }: Props) => (
    <Hint shouldSelect={shouldSelectHint}>
      <Input
        {...props}
        ref={(node) => {
          inputRef(node);
          referenceElementRef(node);
        }}
      />
    </Hint>
  )
);
