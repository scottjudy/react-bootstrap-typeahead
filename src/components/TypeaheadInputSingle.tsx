import React, { RefCallback } from 'react';

import Hint, { ShouldSelect } from './Hint';
import Input from './Input';

import withClassNames from '../behaviors/classNames';

interface TypeaheadInputSingleProps {
  inputRef: RefCallback<HTMLInputElement>;
  referenceElementRef: RefCallback<HTMLInputElement>;
  shouldSelectHint?: ShouldSelect;
}

export default withClassNames(
  ({
    inputRef,
    referenceElementRef,
    shouldSelectHint,
    ...props
  }: TypeaheadInputSingleProps) => (
    <Hint shouldSelect={shouldSelectHint}>
      <Input
        {...props}
        ref={(node) => {
          inputRef(node);
          referenceElementRef(node);
        }}
      />
    </Hint>
  )
);
