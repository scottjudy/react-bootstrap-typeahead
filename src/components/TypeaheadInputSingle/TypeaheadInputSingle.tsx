import React from 'react';

import Hint from '../Hint';
import Input from '../Input';

import { TypeaheadInputProps } from '../../types';
import { propsWithBsClassName } from '../../utils';

const TypeaheadInputSingle = ({
  inputRef,
  referenceElementRef,
  ...props
}: TypeaheadInputProps) => (
  <Hint>
    <Input
      {...propsWithBsClassName(props)}
      ref={(node) => {
        inputRef(node);
        referenceElementRef(node);
      }}
    />
  </Hint>
);

export default TypeaheadInputSingle;
