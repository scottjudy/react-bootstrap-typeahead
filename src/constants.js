export const DEFAULT_LABELKEY = 'label';

export const ALIGN = {
  JUSTIFY: 'justify',
  LEFT: 'left',
  RIGHT: 'right',
};

export const SIZE = {
  LARGE: 'large',
  LG: 'lg',
  SM: 'sm',
  SMALL: 'small',
};
