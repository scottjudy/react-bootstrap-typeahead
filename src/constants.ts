export const DEFAULT_LABELKEY = 'label';
export const SIZES = ['large', 'lg', 'small', 'sm'] as const;
