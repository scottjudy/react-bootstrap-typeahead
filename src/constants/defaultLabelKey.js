export const DEFAULT_LABELKEY = 'label';
