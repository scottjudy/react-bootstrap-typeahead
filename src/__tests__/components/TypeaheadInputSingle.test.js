import { mount } from 'enzyme';
import { head, noop } from 'lodash';
import React from 'react';

import TypeaheadInputSingle from '../../components/TypeaheadInputSingle.react';

import options from '../data';
import { getFormControl, getHint, getInput, keyDown, TestProvider } from '../helpers';
import { TAB } from '../../constants';

describe('<TypeaheadInputSingle>', () => {
  let shouldSelectHint, wrapper;

  beforeEach(() => {
    shouldSelectHint = jest.fn();
    wrapper = mount(
      <TestProvider
        onKeyDown={noop}
        options={options}
        selected={[]}>
        {({ getInputProps }) => (
          <TypeaheadInputSingle
            {...getInputProps()}
            referenceElementRef={noop}
            shouldSelectHint={shouldSelectHint}
          />
        )}
      </TestProvider>
    );
  });

  test('renders a single-select input', () => {
    const input = getFormControl(wrapper);

    expect(input.length).toBe(1);
    expect(input.hasClass('rbt-input')).toBe(true);
    expect(input.hasClass('rbt-input-main')).toBe(true);
  });

  test('displays the selected text', () => {
    const text = 'text';

    wrapper.setProps({ text });

    expect(getInput(wrapper).prop('value')).toBe(text);
  });

  test('displays a hint and calls `shouldSelectHint`', () => {
    const initialItem = head(options);

    wrapper.setProps({
      initialItem,
      isFocused: true,
      isMenuShown: true,
      text: 'Al',
    });

    expect(getHint(wrapper)).toBe(initialItem.name);

    // No need to test the logic for `shouldSelectHint` here; just make sure
    // it's passed through to the `Hint` component and called.
    keyDown(wrapper, TAB);
    expect(shouldSelectHint).toHaveBeenCalledTimes(1);
  });
});
