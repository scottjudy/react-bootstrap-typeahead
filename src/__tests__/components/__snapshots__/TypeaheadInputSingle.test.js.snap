// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeaheadInputSingle> renders a snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "height": "100%",
      "position": "relative",
    }
  }
>
  <input
    aria-autocomplete="both"
    aria-haspopup="listbox"
    autoComplete="off"
    className="rbt-input-main form-control rbt-input"
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="combobox"
    type="text"
    value=""
  />
  <input
    aria-hidden={true}
    className="rbt-input-hint"
    readOnly={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "boxShadow": "none",
        "color": "rgba(0, 0, 0, 0.54)",
        "left": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    tabIndex={-1}
    value=""
  />
</div>
`;
